paimon-core/src/main/java/org/apache/paimon/deletionvectors/DeletionVector.java

```diff
@@ -99,11 +99,11 @@ default boolean checkedDelete(long position) {
      * @return A DeletionVector instance that represents the deserialized data.
      */
     static DeletionVector deserializeFromBytes(byte[] bytes) {
-        try (ByteArrayInputStream bis = new ByteArrayInputStream(bytes);
-                DataInputStream dis = new DataInputStream(bis)) {
-            int magicNum = dis.readInt();
+        try {
+            ByteBuffer buffer = ByteBuffer.wrap(bytes);
+            int magicNum = buffer.getInt();
             if (magicNum == BitmapDeletionVector.MAGIC_NUMBER) {
-                return BitmapDeletionVector.deserializeFromDataInput(dis);
+                return BitmapDeletionVector.deserializeFromByteBuffer(buffer.slice());
```

Comment:
Why is slice needed here

Reply:
This is really just a programming convention to ensure that the bytes passed in are bitmaps only. I take a look that it is already using in `RoaringArray#deserialize`, so it's redundant here.

https://github.com/RoaringBitmap/RoaringBitmap/blob/60dcdb58846037ea8ea712317a0ac887e00fe25b/roaringbitmap/src/main/java/org/roaringbitmap/RoaringArray.java#L544-L550